src/libraries/Common/src/System/Net/SocketAddress.cs

```diff
@@ -93,11 +93,30 @@ public SocketAddress(AddressFamily family, int size)
             }
 
             InternalSize = size;
+
+            // TODO: workaround for https://github.com/dotnet/runtime/issues/33583
+            //       remove when Mono has pinned arrays
+#if MONO
```

Comment:
These libraries do not have Mono-specific configuration. This needs to be either runtime check; or you need to change the build system to introduce Mono-specific configurations for these libraries (there is no prior art).

Reply:
That would explain the failures. I think runtime check is more appropriate, since this would be very temporary.